### DISCUSSION TITLE
Authentication & User data storage

### DISCUSSION THREAD

**Discussion by rakNeoQOWS101** [verdaccio/verdaccio#3609] [Q&A]
Just curious to know the internal working of Verdaccio related to User Authentication & User data storage as i couldn't find any documentation related to it.

Once the user is authenticated, does verdaccio stores the user's cred on the file system where it is hosted or on the verdaccio side as I could see CouchDB ref in the log trace?
I set up verdaccio on the local machine inside the docker container without mounting volume, Logged Into the private registry, and did other operations. Then I killed that container and spawned a new one. As I was already logged into the registry I don't login into it again. And got something like an Authentication token not found error. Then I tried the same thing but after mounting the volume (same storage path as mentioned in the config.yml). After that, I repeated the same steps but didn't get any errors.

So my question is how do we manage the authToken which is created against username and password?

**Comment by rakNeoQOWS101** [comment]
@juanpicado : Can you please help here. Thanks

**Comment by juanpicado** [comment]
> So my question is how do we manage the authToken which is created against username and password?

 I think here you have different issues here, first is (still not documented as you are saying) the token is generated based on a salt secret key, it's unique if you remove your storage, so in this case Docker generate a new storage if you don't mount the first time the volumen, the token you received is based on that secret key which does not exist anymore so next time fails because token is cannot be verified. 

Anyhow, please provide set of steps to reproduce locally and ensure I understood correctly next time. Aside of this I'm improving the docs if you think something is missing feel free to point out where, it's helpful to know where users try to find answers first.
